streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java

```diff
@@ -669,25 +707,19 @@ public void shouldThrowOnNegativeTimeoutForClose() {
 
     @Test
     public void shouldNotBlockInCloseForZeroDuration() throws InterruptedException {
-        final KafkaStreams streams = new KafkaStreams(builder.build(), props);
-        final Thread th = new Thread(() -> streams.close(Duration.ofMillis(0L)));
-
-        th.start();
+        try (final KafkaStreams streams = new KafkaStreams(new StreamsBuilder().build(), props, supplier, time)) {
+            final Thread th = new Thread(() -> streams.close(Duration.ofMillis(0L)));
 
-        try {
+            th.start();
             th.join(30_000L);
```

Comment:
Not sure I follow the logic of this test... We want to show that it doesn't block, but then we just assert that the call doesn't take longer than 30 seconds.

It seems like this test is sensitive to timing, which is not ideal. I guess that what we really wanted to test is that the close method never calls `Thread.sleep()`/`Thread.join()` or maybe just never calls a specific list of blocking methods on the clients. Now that we have all the mocks above, it seems like we can make this assertion directly.

Reply:
I agree, I've actually simplify the logic to just check that `close()` call returns false, since with mock-time it would never elapse any time, and if the call ever waits on the state transition it would not return ever.